src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs

```diff
@@ -144,6 +134,53 @@ static MsQuicApi()
         }
     }
 
+    internal static MsQuicApi AllocateMsQuicApi()
+    {
+        Debug.Assert(IsQuicSupported);
+
+        if (TryLoadMsQuic(out IntPtr msQuicHandle) &&
+            TryOpenMsQuic(msQuicHandle, out QUIC_API_TABLE* apiTable))
+        {
+            return new MsQuicApi(apiTable);
+        }
+
+        throw new Exception("Failed to create MsQuicApi instance");
+    }
+
+    internal static bool TryLoadMsQuic(out IntPtr msQuicHandle) =>
+        NativeLibrary.TryLoad($"{Interop.Libraries.MsQuic}.{MsQuicVersion.Major}", typeof(MsQuicApi).Assembly, DllImportSearchPath.AssemblyDirectory, out msQuicHandle) ||
+        NativeLibrary.TryLoad(Interop.Libraries.MsQuic, typeof(MsQuicApi).Assembly, DllImportSearchPath.AssemblyDirectory, out msQuicHandle);
+
+    internal static bool TryOpenMsQuic(IntPtr msQuicHandle, out QUIC_API_TABLE* apiTable)
```

Comment:
We could alternatively use [LibraryImport] for `MsQuicOpenVersion` and `MsQuicClose`, it might clean up the code a bit at the cost of not being able to unload the lib from the process I think (if we care about that)

Reply:
They already have imports in:
https://github.com/dotnet/runtime/blob/bf47ac0810e53d13a57d10ff1c8b948e5bc4b516/src/libraries/System.Net.Quic/src/System/Net/Quic/Interop/msquic_generated.cs#L2593

I'm not an expert on this, but we control which version of libmsquic we're loading in `TryLoadMsQuic` and we do an attempt to load it that would not fail if the lib is not present. I'm not sure we can achieve the same with the import. Relevant history: https://github.com/dotnet/runtime/pull/49973

Anyway, we might at least use `nameof(MsQuicOpenVersion)` from the generated interop. Though we lived with the string until now, so it's certainly not critical.